external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -87,13 +84,14 @@
 
 
     public KafkaSpout(KafkaSpoutConfig<K, V> kafkaSpoutConfig) {
-        this(kafkaSpoutConfig, new KafkaConsumerFactoryDefault());
+        this.kafkaSpoutConfig = kafkaSpoutConfig;
+        setKafkaConsumerFactory(new KafkaConsumerFactoryDefault());
     }
     
-    //This constructor is here for testing
-    KafkaSpout(KafkaSpoutConfig<K, V> kafkaSpoutConfig, KafkaConsumerFactory<K, V> kafkaConsumerFactory) {
-        this.kafkaSpoutConfig = kafkaSpoutConfig;                 // Pass in configuration
+    //This setter is here for testing
+    final KafkaSpout setKafkaConsumerFactory(KafkaConsumerFactory<K, V> kafkaConsumerFactory) {
```

Comment:
I would rather leave this with the constructor. We shouldn't be making this class mutable unless we have a very good reason to do so.